src/hooks/usePendingSwapData.ts

```diff
@@ -335,7 +339,7 @@ async function fetchAndPopulatePendingSwaps(
 
   // Step 3: fetch aync data via bridge contract
   let fetchedPendingSwaps = await Promise.all(
```

Comment:
While you're at it, could you take a look and change these to
```
const events = (
    await Promise.all(
      eventFilters.map((filter) =>
        bridgeContract.queryFilter(filter, startBlock, "latest"),
      ),
    )
  ).flat()

  // Step 3: fetch aync data via bridge contract
  const fetchedPendingSwaps = (
    await Promise.all(
      events.map((event) =>
        fetchPendingSwapInfo(bridgeContract, event, chainId),
      ),
    )
  ).filter(Boolean)
  if (fetchedPendingSwaps.length === 0) return
  ```
Reassignments are usually not a good idea.

Reply:
This is a good call but can you make an issue on this and make the change in a separate PR for the sake of keeping the scope of this to removal of POOL/TOKENS_MAP?